### PR TITLE
Task_8

### PR DESCRIPTION
**Pull request by Xodoc**

### CODE REVIEW

File: ConsoleApp/Shared/appsettings.json [hunk 1]
```diff
@@ -16,5 +16,6 @@
   "MaxDays": 7,
   "MinDays": 0,
   "Hours": 12,
-  "IsDebug": false
+  "IsDebug": false,
+  "SpecifiedTime": 200
```
Comment (denja2) [inline ConsoleApp/Shared/appsettings.json:20]: p3 bad name

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 2]
```diff
@@ -70,32 +71,38 @@ public async Task<List<TemperatureInfo>> GetTemperaturesAsync(IEnumerable<string
...
-
                     stopwatch.Stop();
-                }
-                else
-                {
-                    weather = await GetWeatherAsync(name);
-                }
 
-                var info = new TemperatureInfo();
+                    if (stopwatch.ElapsedMilliseconds > _config.SpecifiedTime)
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:86]: P1 Token cancellation should be called outside the operation

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 3]
```diff
@@ -70,32 +71,38 @@ public async Task<List<TemperatureInfo>> GetTemperaturesAsync(IEnumerable<string
...
-
-                if (_config.IsDebug)
+                await cityNames.ParallelForEachAsync(async name =>
                 {
-                    stopwatch.Start();
+                    var stopwatch = new Stopwatch();
+                    var weather = new CurrentWeather();
 
+                    stopwatch.Start();
                     weather = await GetWeatherAsync(name);
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:83]: P1 Get weather should support Cncelation token

**Review by denja2 — CHANGES_REQUESTED** [review]

File: ConsoleApp/Shared/appsettings.json (hunk 1 above)
Comment (Xodoc) [reply ConsoleApp/Shared/appsettings.json:20]: done

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs (hunk 2 above)
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/WeatherRepository.cs:86]: done

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs (hunk 3 above)
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/WeatherRepository.cs:83]: done

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 4]
```diff
@@ -25,7 +26,7 @@ public WeatherRepository(IConfiguration config)
             _config = config;
         }
 
-        public async Task<CurrentWeather> GetWeatherAsync(string cityName)
+        public async Task<CurrentWeather> GetWeatherAsync(string cityName, CancellationTokenSource cts)
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:29]: Use token

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 5]
```diff
@@ -36,6 +37,11 @@ public async Task<CurrentWeather> GetWeatherAsync(string cityName)
 
                 var weather = await responseMessage.Content.ReadAsStringAsync();
 
+                if (cts.IsCancellationRequested == true)
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:40]: pass token in all async methods

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs (hunk 4 above)
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/WeatherRepository.cs]: done

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs (hunk 5 above)
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/WeatherRepository.cs]: done

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 6]
```diff
@@ -36,6 +37,11 @@ public async Task<CurrentWeather> GetWeatherAsync(string cityName)
 
                 var weather = await responseMessage.Content.ReadAsStringAsync();
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:38]: token

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 7]
```diff
@@ -36,6 +37,11 @@ public async Task<CurrentWeather> GetWeatherAsync(string cityName)
...
 
+                if (ct.IsCancellationRequested == true)
+                {
+                    ct.ThrowIfCancellationRequested();
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:42]: no need if

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 8]
```diff
@@ -66,36 +72,37 @@ public async Task<WeatherForecast> GetWeatherForecastAsync(string cityName, int
...
+                    weather = await GetWeatherAsync(name, ct);
                     stopwatch.Stop();
-                }
-                else
-                {
-                    weather = await GetWeatherAsync(name);
-                }
 
-                var info = new TemperatureInfo();
+                    if (ct.IsCancellationRequested == true)
```
Comment (denja2) [inline ConsoleApp/DAL/Repositories/WeatherRepository.cs:90]: no need

**Pull request merged**